pkg/kubelet/cm/cpumanager/topology/topology.go

```diff
@@ -260,9 +332,10 @@ func Discover(machineInfo *cadvisorapi.MachineInfo) (*CPUTopology, error) {
 			if coreID, err := getUniqueCoreID(core.Threads); err == nil {
 				for _, cpu := range core.Threads {
 					CPUDetails[cpu] = CPUInfo{
-						CoreID:     coreID,
-						SocketID:   core.SocketID,
-						NUMANodeID: node.Id,
+						CoreID:        coreID,
+						SocketID:      core.SocketID,
+						NUMANodeID:    node.Id,
+						UnCoreCacheID: core.UncoreCaches[0].Id,
```

Comment:
this can break in some server-grade CPUs which don't expose uncore cache. I suggest to replace with something like
```
const (
	NullUncoreCacheID int = 65535 // TODO: "high enough" value to signal "impossible"
)

func getUncoreCacheID(uncoreCaches []cadvisorapi.Cache) int {
	if len(uncoreCaches) < 1 {
		return NullUncoreCacheID
	}
	// Even though cadvisor API returns a slice, we only expect either 0 or 1 uncore caches,
	// so everything past first entry should be discarded
	return uncoreCaches[0].Id
}

// ...

UnCoreCacheID: getUncoreCacheID(core.UncoreCaches),
```

we may want to use a better ID to signal "no uncore caches detected

Reply:
I'm not totally sure why, but can't we presume it will be `0`? I think that's the value found in `/sys/device/system/cpu/cpuX/cache/index3/id`

Reply:
seen myself that in some cases (= with some commonly available xeons) `cadvisor` just gives us an empty slice, which will result in nil pointer deference.

I usually prefer fixing the bug in the right layer rather than add defensive programming, but engineering it's all about tradeoffs, so in this case I'd start adding in our code some helper function like I described, and then (in parallel/later) go review and fix `cadvisor`.

Reply:
I agree, for now I added the function but if we see an empty slice I return `0`

Reply:
After reviewing this should fallback to the socketid or to the node.caches[0] value. For now we left it at the socketID.

Reply:
Added a test case to handle empty UncoreCacheIDs in the Discover function

Reply:
using SocketID as fallback SGTM, thanks!